### PR TITLE
List Menu Dual Highlight Bug Fix + Shop Screen Sell DRG Alignment Fix

### PR DESCRIPTION
**Pull request by Ink230**
## List Menu Bug Fix - #294 

• Refactored render code to be simpler with sharing one highlighter, one selectSlot
• Refactored vertical input handling to be as straightforward as a shared state dual pane distinct menu system can be
• Added logic that was missing to the Left and Right toggles of state 

• This may have conflicts with #268 (which did not address this bug)

## Equipment => Armed

• Equipment broke out of the boundaries on the Sell => Equipment/Items menu
• Armed is the original name and Armed is used elsewhere in the Menu system
• I did not restore the original menu in the mid left of the screen but kept the one already present
• See following screenshots for details

### Original Bug
image

### What Retail Looks Like
image

### What The Fix Looks Like
image

### CODE REVIEW

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java [hunk 1]
```diff
@@ -124,18 +120,20 @@ protected void render() {
...
           if(MathHelper.inBox(this.mouseX, this.mouseY, 186, 40, 174, 122)) {
             if(this.slotScrollItem < gameState_800babc8.itemCount_1e6.get() - 7) {
               playSound(1);
               this.slotScrollItem++;
-              this.setCurrent(gameState_800babc8.items_2e9, this.items, this.slotScrollItem + this.selectedSlotItem);
+              this.setCurrent(gameState_800babc8.items_2e9, this.items, this.slotScrollItem + this.selectedSlot);
             }
           }
         }
+
```
Comment (LordMonoxide) [inline src/main/java/legend/game/inventory/screens/ItemListScreen.java:135]: Extra new lines (try to keep changes like this to a minimum to simplify file history)

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java [hunk 2]
```diff
@@ -152,6 +150,30 @@ protected void render() {
     }
   }
 
+  private void handleVerticalInput(final boolean isScrollingUp)
+  {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/inventory/screens/ItemListScreen.java:154]: CS - brace on same line

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java [hunk 3]
```diff
@@ -198,20 +220,20 @@ protected void mouseMove(final int x, final int y) {
...
     if(this.loadingStage == 1) {
       for(int i = 0; i < Math.min(7, gameState_800babc8.equipmentCount_1e4.get() - this.slotScrollEquipment); i++) {
-        if(this.selectedSlotEquipment != i && MathHelper.inBox(x, y, 8, 31 + FUN_800fc814(i), 174, 17)) {
+        if(this.selectedSlot != i && MathHelper.inBox(x, y, 8, 31 + FUN_800fc814(i), 174, 17)) {
           playSound(1);
-          this.selectedSlotEquipment = i;
-          this.equipmentHighlight.y_44 = FUN_800fc814(i) + 32;
-          this.setCurrent(gameState_800babc8.equipment_1e8, this.equipment, this.slotScrollEquipment + this.selectedSlotEquipment);
+          this.selectedSlot = i;
+//          this.equipmentHighlight.y_44 = FUN_800fc814(i) + 32;
```
Comment (LordMonoxide) [inline src/main/java/legend/game/inventory/screens/ItemListScreen.java:226]: Commented code? Are these needed or should they be removed?

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java [hunk 4]
```diff
@@ -46,13 +46,11 @@ public class ItemListScreen extends MenuScreen {
   private double scrollAccumulator;
   private final Runnable unload;
 
-  private int selectedSlotEquipment;
-  private int selectedSlotItem;
+  private int selectedSlot;
   private int slotScrollEquipment;
```
Comment (LordMonoxide) [inline src/main/java/legend/game/inventory/screens/ItemListScreen.java:50]: Should the scroll variables be combined as well, or is there a reason to keep them separate?

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

**Comment by LordMonoxide** [comment]
Can you repoint this to RC? I want to get it in for the recommended build

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java (hunk 4 above)
Comment (Ink230) [reply src/main/java/legend/game/inventory/screens/ItemListScreen.java:50]: I believe they're necessary. Each menu tracks its own independent state of how far down either is scrolled. You need something to say Equipment has scrolled down 9 items while Items has scrolled down 30 items (aka, holding state) for the way the lists are rendered / pulled in from gameState. They are basically just indices of two distinct sets. And the external side-effects of some decompiled functions are basically dependencies we're working to satisfy that use these indices.

This implementation, which was the original code, makes it easier (or possible) to navigate between the two menus freely with Left and Right inputs. Refactoring this into a better design pattern seems unjustified for the scope of the change, but may be justified if the other menus are all refactored into a better pattern.

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java (hunk 3 above)
Comment (Ink230) [reply src/main/java/legend/game/inventory/screens/ItemListScreen.java:226]: Unimplemented mouse support changes. Working on the implementations for a commit.

File: src/main/java/legend/game/inventory/screens/ItemListScreen.java (hunk 4 above)
Comment (Ink230) [reply src/main/java/legend/game/inventory/screens/ItemListScreen.java:50]: Notice in the Goods menu that the scroll is global between both menus when using one slotScroll variable. Scrolling in one, scrolls in the other. This is not ideal behaviour.

**Comment by Ink230** [comment]
Mouse input and functionality is restored. Keyboard navigation was what was worked on previously.

Added full functionality in navigation and selection for mouse and keyboard distinctly. Circle or back functionality on Discard menu, and in general, the menus currently, don't exist. This is a change outside of the scope of this PR. Selecting "No" is the only option to go back currently; there are no UI soft locks.

Controller support can be added into the keyPress method (and potentially renamed) as all actions are 1:1 to controllers. Easiest way is to add additional cases on the keyPress event. Other implementations might struggle with lack of controller support in the overall Menu system, see PR #268.

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**